app/post_acknowledgements.go

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package app
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+
+	"github.com/mattermost/mattermost-server/v6/app/request"
+	"github.com/mattermost/mattermost-server/v6/model"
+	"github.com/mattermost/mattermost-server/v6/shared/mlog"
+	"github.com/mattermost/mattermost-server/v6/store"
+)
+
+func (a *App) SaveAcknowledgementForPost(c *request.Context, userID, postID string) (*model.PostAcknowledgement, *model.AppError) {
+	post, err := a.GetSinglePost(postID, false)
+	if err != nil {
+		return nil, err
+	}
+
+	channel, err := a.GetChannel(c, post.ChannelId)
+	if err != nil {
+		return nil, err
+	}
+
+	if channel.DeleteAt > 0 {
+		return nil, model.NewAppError("SaveAcknowledgementForPost", "api.acknowledgement.save.archived_channel.app_error", nil, "", http.StatusForbidden)
+	}
+
+	acknowledgedAt := model.GetMillis()
+	acknowledgement, nErr := a.Srv().Store().PostAcknowledgement().Save(userID, postID, acknowledgedAt)
+
+	if nErr != nil {
+		var appErr *model.AppError
+		switch {
+		case errors.As(nErr, &appErr):
+			return nil, appErr
+		default:
+			return nil, model.NewAppError("SaveAcknowledgementForPost", "app.acknowledgement.save.save.app_error", nil, "", http.StatusInternalServerError).Wrap(nErr)
+		}
+	}
+
+	a.Srv().Go(func() {
+		a.sendAcknowledgementEvent(model.WebsocketEventAcknowledgementAdded, acknowledgement, post)
+	})
+
+	return acknowledgement, nil
+}
+
+func (a *App) DeleteAcknowledgementForPost(c *request.Context, userID, postID string) *model.AppError {
+	post, err := a.GetSinglePost(postID, false)
+	if err != nil {
+		return err
+	}
+
+	channel, err := a.GetChannel(c, post.ChannelId)
+	if err != nil {
+		return err
+	}
+
+	if channel.DeleteAt > 0 {
+		return model.NewAppError("DeleteAcknowledgementForPost", "api.acknowledgement.delete.archived_channel.app_error", nil, "", http.StatusForbidden)
+	}
+
+	oldAck, nErr := a.Srv().Store().PostAcknowledgement().Get(userID, postID)
+
+	if nErr != nil {
+		var nfErr *store.ErrNotFound
+		switch {
+		case errors.As(nErr, &nfErr):
+			return model.NewAppError("GetPostAcknowledgement", "app.acknowledgement.get.app_error", nil, "", http.StatusNotFound).Wrap(nErr)
+		default:
+			return model.NewAppError("GetPostAcknowledgement", "app.acknowledgement.get.app_error", nil, "", http.StatusInternalServerError).Wrap(nErr)
+		}
+	}
+
+	if model.GetMillis()-oldAck.AcknowledgedAt > 5*60*1000 {
```

Comment:
is the 5 min configurable or is it supposed to be hardcoded?

Reply:
It's supposed to be hardcoded.
cc @matthewbirtch

Reply:
@matthewbirtch oh sorry this one is the time the user has to un-acknowledge a post, not for persistent notifications

Reply:
Oh, my mistake, sorry it was hard to tell what this was referring to.

Yes, this wasn't something we identified as configurable so hardcoded I guess is the way to go